src/Android/Accessibility/AccessibilityHelpers.cs

```diff
@@ -515,24 +530,23 @@ public static int GetNodeHeight(AccessibilityNodeInfo node)
             return nodeRectHeight;
         }
 
-        private static int GetStatusBarHeight()
+        private static int GetStatusBarHeight(AccessibilityService service)
         {
-            return GetSystemResourceDimenPx("status_bar_height");
+            return GetSystemResourceDimenPx(service, "status_bar_height");
         }
 
-        private static int GetNavigationBarHeight()
+        private static int GetNavigationBarHeight(AccessibilityService service)
         {
-            return GetSystemResourceDimenPx("navigation_bar_height");
+            return GetSystemResourceDimenPx(service, "navigation_bar_height");
         }
 
-        private static int GetSystemResourceDimenPx(string resName)
+        private static int GetSystemResourceDimenPx(AccessibilityService service, string resName)
         {
-            var activity = (MainActivity)CrossCurrentActivity.Current.Activity;
             var barHeight = 0;
-            var resourceId = activity.Resources.GetIdentifier(resName, "dimen", "android");
+            var resourceId = service.Resources.GetIdentifier(resName, "dimen", "android");
             if (resourceId > 0)
             {
-                barHeight = activity.Resources.GetDimensionPixelSize(resourceId);
+                barHeight = service.Resources.GetDimensionPixelSize(resourceId);
```

Comment:
To save allocation on the heap you can simply return the result of this call, and below the fall-back would be to return zero explicitly (vs. using a variable for a single possible assignment).

Reply:
Done